src/components/LinkChooser/state/machine.ts

```diff
@@ -0,0 +1,233 @@
+    "updateQueryFromObject",
+    "updateCustomLink",
+    "setSelectedSearchResult",
+    "emitSelectSearchResult",
+];
+
+const sharedActions = {
+    CLEARING: {
+        actions: ["clearSelectedResult", "updateQueryFromString", "emitSelectSearchResult"],
+    },
+    OPEN_PREVIEW: {
+        actions: ["openPreview"],
+    },
+    COPY_TO_CLIPBOARD: {
+        actions: ["copyLinkToClipboard"],
+    },
+};
+
+const initializeSectionState = (
+    initial: string,
+    id: string,
+    src: (context: LinkChooserContext) => Promise<LinkChooserEventData>,
+) => ({
+    initial,
+    states: {
+        [SectionState.Loaded]: {
+            on: {
+                ...typingAction,
+            },
+        },
+        [SectionState.Typing]: {
+            after: {
+                [DEBOUNCE_TIMEOUT]: { target: SectionState.Fetching },
+            },
+            on: {
+                ...typingAction,
+            },
+        },
+        [SectionState.Fetching]: {
+            invoke: {
+                id,
+                src,
+                onDone: [
+                    {
+                        target: SectionState.Loaded,
+                        actions: ["updateDropdownSearchResults", "updateCustomLink", "resolveFetching"],
+                        cond: "isDefaultSection",
+                    },
+                    {
+                        target: SectionState.Loaded,
+                        actions: ["updateDropdownSearchResults"],
+                    },
+                ],
+                onError: SectionState.Error,
+            },
+            on: {
+                ...typingAction,
+            },
+        },
+        [SectionState.Error]: {
+            on: {
+                ...typingAction,
+            },
+        },
+    },
+});
+
+export const linkChooserMachine = createMachine<LinkChooserContext, DoneInvokeEvent<LinkChooserEventData>>(
+    {
+        id: "link-chooser",
+        initial: LinkChooserState.Idle,
+        states: {
+            [LinkChooserState.Idle]: {
+                on: {
+                    OPEN_DROPDOWN: [
+                        {
+                            target: LinkChooserState.Focused,
+                            actions: ["populateDropdownSearchResultsWithRecentQueries"],
+                            cond: "isQueryEmpty",
+                        },
+                        {
+                            target: LinkChooserState.Focused,
+                            actions: ["updateCustomLink"],
+                        },
+                    ],
+                    ...sharedActions,
+                },
+            },
+            [LinkChooserState.Focused]: {
+                initial: DropdownState.Default,
+                states: {
+                    [DropdownState.Default]: {
+                        ...initializeSectionState(SectionState.Loaded, "fetchGlobal", fetchGlobalSearchResults),
+                        on: {
+                            GO_TO_GUIDELINES: {
```

Comment:
you can directly provide the next target, just provide `DropdownState.Guidelines` directly

Reply:
done 👍